rpc/client.go

```diff
@@ -109,6 +135,84 @@ func (opts *UploadReportOptions) artifactConsumer(ctx context.Context, testChan
 	}
 }
 
+func getSignedURL(task string, execution int, awsRegion string, data []byte, awsSecretKey string, awsAccessKey string, resultsHandlerHost string, resultType string) (string, error) {
+	service := "execute-api"
+	if awsSecretKey == "" || awsAccessKey == "" || resultsHandlerHost == "" || resultType == "" {
+		return "", errors.New("Getting signed URL failed. AWS secret key, AWS access key, resuls handler host and result type required.")
+	}
+	client := &http.Client{}
+	name := uuid.New()
+	url := fmt.Sprintf("%s/evergreen/%s/%s/%s/%s", resultsHandlerHost, task, strconv.Itoa(execution), resultType, name.String())
+	req, err := http.NewRequest(http.MethodPut, url, bytes.NewBuffer(data))
+	if err != nil {
+		return "", err
+	}
+	aws_credentials := credentials.NewStaticCredentials(awsAccessKey, awsSecretKey, "")
+	signer := v4.NewSigner(aws_credentials)
+	_, err = signer.Sign(req, nil, service, awsRegion, time.Now())
+	if err != nil {
+		return "", err
+	}
+	response, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	body, error := ioutil.ReadAll(response.Body)
+	if error != nil {
+		return "", err
+	}
+	// close response body
+	response.Body.Close()
+	var response_body SignedUrl
+	err = json.Unmarshal(body, &response_body)
+	if error != nil {
+		return "", err
+	}
+	grip.Info(message.Fields{
+		"request_url": url,
+		"signed_url":  response_body.Signed_Url,
+	})
+	return response_body.Signed_Url, nil
+}
+
+func uploadTestReport(signedUrl string, data []byte) error {
+	client := &http.Client{}
+	req, err := http.NewRequest(http.MethodPut, signedUrl, bytes.NewBuffer(data))
+	if err != nil {
+		return err
+	}
+	_, err = client.Do(req)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// uploadResults uploads cedar results to S3 bucket which triggers rollups in perf-summarizer-service(PSS)
+func uploadResultsToPSS(report *poplar.Report, awsSecretKey string, awsAccessKey string, resultsHandlerHost string, resultType string, dryRun bool) error {
```

Comment:
The Report [structure](https://pkg.go.dev/github.com/evergreen-ci/poplar#hdr-Report) should include a `bucket` top-level key holding the user-submitted keys. I think we should remove those from the report json before sending it; we have no need of those keys or the bucket information by that point and shouldn't submit user creds. The `TestArtifact` struct will have the artifact-specific buckets and prefixes that the argo workflow can consume.